src/notification.c

```diff
@@ -382,6 +363,122 @@ check_notification (GVariant *notification,
   return TRUE;
 }
 
+static gboolean
+validate_icon_more (GVariant *v)
+{
+  g_autoptr(GIcon) icon = g_icon_deserialize (v);
+  GBytes *bytes;
+  g_autofree char *name = NULL;
+  const char *argv[] = { LIBEXECDIR "/xdg-desktop-portal-validate-icon", "name", NULL };
+  xdp_autofd int fd = -1;
+  g_autoptr(GOutputStream) stream = NULL;
+  gssize written;
+  int status;
+  g_autofree char *err = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (G_IS_THEMED_ICON (icon))
+    {
+      g_autofree char *a = g_strjoinv (" ", (char **)g_themed_icon_get_names (G_THEMED_ICON (icon)));
+      g_debug ("Icon validation: themed icon (%s) is ok", a);
+      return TRUE;
+    }
+
+  if (!G_IS_BYTES_ICON (icon))
+    {
+      g_warning ("Unexpected icon type: %s", G_OBJECT_TYPE_NAME (icon));
+      return FALSE;
+    }
+
+  bytes = g_bytes_icon_get_bytes (G_BYTES_ICON (icon));
+  fd = g_file_open_tmp ("iconXXXXXX", &name, &error); 
+  if (fd == -1)
+    {
+      g_debug ("Icon validation: %s", error->message);
+      return FALSE;
+    }
+
+  stream = g_unix_output_stream_new (fd, TRUE);
+  written = g_output_stream_write_bytes (stream, bytes, NULL, &error);
```

Comment:
From the default session bus config:

  <!-- For the session bus, override the default relatively-low limits
       with essentially infinite limits, since the bus is just running
       as the user anyway, using up bus resources is not something we need
       to worry about. In some cases, we do set the limits lower than
       "all available memory" if exceeding the limit is almost certainly a bug,
       having the bus enforce a limit is nicer than a huge memory leak. But the
       intent is that these limits should never be hit. -->

  <!-- the memory limits are 1G instead of say 4G because they can't exceed 32-bit signed int max -->
  <limit name="max_incoming_bytes">1000000000</limit>
  <limit name="max_incoming_unix_fds">250000000</limit>
  <limit name="max_outgoing_bytes">1000000000</limit>
  <limit name="max_outgoing_unix_fds">250000000</limit>
  <limit name="max_message_size">1000000000</limit>

Can our dbus proxy enforce a smaller limit ?

Reply:
From what I can tell the proxy has no limits itself, so that probably isn't ideal.

Reply:
Right. I'm saying: if we wanted to address bus DOS, the proxy would be the place to do it